src/main/java/internet/shop/dao/impl/CarDaoImpl.java

```diff
@@ -1,54 +1,180 @@
+                + " d.driver_id as d_did, d.driver_name as d_dn, d.driver_license_number as d_dln"
+                + " from cars c"
+                + " inner join cars_drivers cd on c.car_id = cd.car_id"
+                + " inner join drivers d on d.driver_id = cd.driver_id"
+                + " where c.car_id = ? and c.car_deleted = false;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            optionalCar = Optional.of(getCarFromDB(resultSet));
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't get the data by id: " + id, exception);
+        }
+        return optionalCar;
     }
 
     @Override
     public List<Car> getAll() {
-        return Storage.getCars();
+        List<Car> carList = new ArrayList<>();
+        String query = "SELECT car_id from cars where car_deleted = false";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                Optional<Car> optionalCar = get(resultSet.getObject("car_id", Long.class));
+                optionalCar.ifPresent(carList::add);
+            }
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't get all data from DB", exception);
+        }
+        return carList;
     }
 
     @Override
     public Car update(Car car) {
-        List<Car> dbCar = Storage.getCars();
-        IntStream.range(0, dbCar.size())
-                .filter(i -> dbCar.get(i).equals(car))
-                .findFirst()
-                .ifPresent(i -> dbCar.set(i, car));
+        String queryForDeleteInCarsDrivers = "delete from cars_drivers where car_id = ?;";
+        String queryForUpdateInCars = "update cars set car_deleted = true where car_id = ?;";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteStmtCarsDrivers = connection
+                        .prepareStatement(queryForDeleteInCarsDrivers);
+                PreparedStatement updateStmtCars = connection
+                        .prepareStatement(queryForUpdateInCars)) {
+            deleteStmtCarsDrivers.setLong(1, car.getId());
+            updateStmtCars.setLong(1, car.getId());
+            deleteStmtCarsDrivers.executeUpdate();
+            updateStmtCars.executeUpdate();
+        } catch (SQLException exception) {
+            throw new DataProcessingException("Can't update the data - "
+                    + car + "in table of cars or cars_drivers", exception);
+        }
+        create(car);
```

Comment:
this way the car's id is changed - let's not change it and update the existing car (with the id of the car that is passed to the method)

Reply:
done